pkg/controller/portworxdiag/pod.go

```diff
@@ -0,0 +1,135 @@
+package portworxdiag
+
+import (
+	"fmt"
+	"os"
+
+	portworxv1 "github.com/libopenstorage/operator/pkg/apis/portworx/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	PortworxDiagLabel = "portworx-diag"
+)
+
+func v1Volume(name, path string, hpType v1.HostPathType) v1.Volume {
+	return v1.Volume{
+		Name: name,
+		VolumeSource: v1.VolumeSource{
+			HostPath: &v1.HostPathVolumeSource{
+				Path: path,
+				Type: &hpType,
+			},
+		},
+	}
+}
+
+func volumes() []v1.Volume {
+	return []v1.Volume{
+		v1Volume("dockersock", "/var/run/docker.sock", v1.HostPathUnset),
```

Comment:
Let's check if we need all these volumes and mounts in the diags pod. You can create a separate ticket and then do it later.

Reply:
https://portworx.atlassian.net/browse/PWX-32179